trestle/core/commands/author/component.py

```diff
@@ -0,0 +1,275 @@
+        self.add_argument('-o', '--output', help=const.HELP_MARKDOWN_NAME, required=True, type=str)
+
+    def _run(self, args: argparse.Namespace) -> int:
+        try:
+            log.set_log_level_from_args(args)
+
+            return self.component_generate_all(args.trestle_root, args.name, args.profile, args.output, {}, None, False)
+
+        except Exception as e:  # pragma: no cover
+            return handle_generic_command_exception(e, logger, 'Generation of the component markdown failed')
+
+    def component_generate_all(
+        self,
+        trestle_root: pathlib.Path,
+        comp_def_name: str,
+        profile_name: str,
+        markdown_dir_name: str,
+        yaml_header: Dict[str, Any],
+        sections_dict: Optional[Dict[str, str]],
+        overwrite_header_values: bool
+    ) -> int:
+        """Generate markdown for all components in comp def."""
+        if not file_utils.is_directory_name_allowed(markdown_dir_name):
+            raise TrestleError(f'{markdown_dir_name} is not an allowed directory name')
+        md_path = trestle_root / markdown_dir_name
+        md_path.mkdir(parents=True, exist_ok=True)
+        component_def, _ = load_validate_model_name(trestle_root, comp_def_name, comp.ComponentDefinition)
+        rc = CmdReturnCodes.SUCCESS.value
+        for component in as_list(component_def.components):
+            rc = self.component_generate_by_name(
+                trestle_root,
+                comp_def_name,
+                component.title,
+                profile_name,
+                md_path / component.title,
+                yaml_header,
+                sections_dict,
+                overwrite_header_values
+            )
+            if rc != CmdReturnCodes.SUCCESS.value:
+                break
+        return rc
+
+    def component_generate_by_name(
+        self,
+        trestle_root: pathlib.Path,
+        comp_def_name: str,
+        comp_name: str,
+        profile_name: str,
+        markdown_dir_path: pathlib.Path,
+        yaml_header: dict,
+        sections_dict: Optional[Dict[str, str]],
+        overwrite_header_values: bool
+    ) -> int:
+        """Create markdown based on the component and profile."""
+        profile_path = ModelUtils.full_path_for_top_level_model(trestle_root, profile_name, prof.Profile)
+        profile_resolver = ProfileResolver()
+        resolved_catalog = profile_resolver.get_resolved_profile_catalog(trestle_root, profile_path)
+        catalog_interface = CatalogInterface(resolved_catalog)
+        comp_def, _ = load_validate_model_name(trestle_root, comp_def_name, comp.ComponentDefinition)
+        context = ControlContext.generate(ContextPurpose.COMPONENT, True, trestle_root, markdown_dir_path)
+        context.sections_dict = sections_dict
+        context.prompt_responses = True
+        context.overwrite_header_values = overwrite_header_values
+        context.comp_def = comp_def
```

Comment:
This is stylistic, but instead of calling a function with many parameters I am systematically adding state to a context object and then calling the function with that context.

Reply:
I fixed a small bug in the test you mentioned and enhanced the test, including a check that md file size does not change on second generation.